docker/linter/Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Use an Alpine-based Python image
+FROM python:3.11-alpine
+
+ARG WORKDIR=/usr/src/app
+ARG CHECKSTYLE_VERSION=10.12.3
+
+USER root
+
+RUN apk --no-cache add bash curl openjdk11-jdk nodejs npm && \
+  wget -O /opt/checkstyle.jar https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${CHECKSTYLE_VERSION}/checkstyle-${CHECKSTYLE_VERSION}-all.jar && \
+  echo -e '#!/bin/sh\njava -jar /opt/checkstyle.jar "$@"' > /usr/local/bin/checkstyle && \
+  chmod +x /usr/local/bin/checkstyle
+
+WORKDIR ${WORKDIR}
+
+# Copy linter dependencies
+COPY ./docker/linter/linter-requirements.txt ${WORKDIR}/requirements.txt
```

Comment:
can we delete this line as well as the file? i assume we don't need since we're using the ones in dev-reqs/reqs